cmake/modules/extensions.cmake

```diff
@@ -4613,12 +4607,21 @@ function(zephyr_linker_dts_memory)
   endif()
   zephyr_string(SANITIZE name ${name})
 
-  zephyr_linker_memory(
-    NAME  ${name}
-    START ${addr}
-    SIZE  ${size}
-    FLAGS ${DTS_MEMORY_FLAGS}
-  )
+  dt_prop(flags PATH ${DTS_MEMORY_PATH} PROPERTY "zephyr,memory-region-flags")
+  if(NOT DEFINED flags)
+    zephyr_linker_memory(
+      NAME  ${name}
+      START ${addr}
+      SIZE  ${size}
+    )
```

Comment:
this actually change behavior of all existing code, because devicetrees today are not having memory-region-flags set, and thus defaults to `rw` in current implementation on main.

With this PR, this behavior changes to empty, meaning it will follow the rules of the linker.
Perhaps it would be better to keep current behavior, so that missing `memory-region-flags` means default to `rw` (existing behavior), and then nodes which don't want to define actual flags can then use an empty string, like this:
```
zephyr,memory-region = "MY@OTHER@NAME";
zephyr,memory-region-flags = "";
```

Thoughts ?

Reply:
I got it. That's the way it should be.
I fixed this as such and added a test item.